components/CategoryFilter.js

```diff
@@ -0,0 +1,26 @@
+import useSWR from "swr";
+
+export default function CategoryFilter({ onSelectCategory }) {
+  const { data: categories, isLoading, error } = useSWR("api/categories");
```

Comment:
```suggestion
  const { data: categories, isLoading, error } = useSWR("/api/categories");
```
usually SWR expects a leading `/` if it’s a relative path. Without the `/` it will be relative to the current route, which may cause 404...